src/read/line.rs

```diff
@@ -1667,6 +1670,35 @@ where
     pub fn md5(&self) -> &[u8; 16] {
         &self.md5
     }
+
+    /// The source code of this file. (UTF-8 source text string with "\n" line
+    /// endings). Note that this may return an empty attribute that indicates
+    /// that no source code is available. The `FileEntry::source` function
+    /// handles this automatically.
+    ///
+    /// This is an accepted proposal for DWARFv6 (Issue 180201.1: DWARF and
+    /// source text embedding)
+    pub fn source_attr(&self) -> Option<AttributeValue<R, Offset>> {
+        self.source.clone()
+    }
+
+    /// The source code of this file. (UTF-8 source text string with "\n" line
+    /// endings)
+    pub fn source(&self, unit: &crate::UnitRef<'_, R>) -> crate::Result<Option<R>> {
```

Comment:
I can see why you've done this, but I prefer to avoid dependencies on `UnitRef` or `Unit` here. Currently nothing outside of the `read::dwarf` module requires them. I'd especially like to avoid this since the DWARF 6 support won't need it. I'd be fine with simply documenting that if the DWARF version is 5 then empty may mean not present, and let the user deal with it.